src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -2051,11 +2052,13 @@ namespace cryptonote
       unsigned int b = 0;
       const time_t time_boundary = now - static_cast<time_t>(seconds[n]);
       for (time_t ts: timestamps) b += ts >= time_boundary;
-      const double p = probability(b, seconds[n] / DIFFICULTY_TARGET_V2);
-      MDEBUG("blocks in the last " << seconds[n] / 60 << " minutes: " << b << " (probability " << p << ")");
+      const double dur = seconds[n] / DIFFICULTY_TARGET_V2; // duration of block generation we are measuring
+      const double p = probability(b, dur);
+      MDEBUG("blocks in the last " << seconds[n] / 60 << " minutes: " << b << " (probability actual " << p << ", expected >=" << threshold << ")");
       if (p < threshold)
       {
-        MWARNING("There were " << b << (b == max_blocks_checked ? " or more" : "") << " blocks in the last " << seconds[n] / 60 << " minutes, there might be large hash rate changes, or we might be partitioned, cut off from the Monero network or under attack, or your computer's time is off. Or it could be just sheer bad luck.");
+        const long b_min_exp = std::lround(threshold * dur); // lower bound on blocks we expect during measurement
+        MWARNING("There were " << b << (b == max_blocks_checked ? " or more" : "") << " blocks in the last " << seconds[n] / 60 << " minutes, while we expected >=" << b_min_exp << ". There might be large hash rate changes, we might be partitioned, or your computer's time is off.");
 
```

Comment:
Part of the message got lost in the patch.

Reply:
He/she previously said it was intentional since the original message was redundant, I assume between  "there might be large hash rate changes" and "Or it could be just sheer bad luck."

Reply:
Hello. Yes, I took the liberty of shortening the message. I felt that the parts I removed were either redundant or unclear.

Reply:
The under attack part got lost and is not redundant at first glance. If you really want it out I won't insist, but it does remove the point of telling the user something fishy might be going on (even though more unlikely that the other possibilities).

Reply:
Ok, I returned it.